### PR TITLE
Use diversification labels when performing CDI derivation or keygen

### PR DESCRIPTION
**Pull request by bluegate010**

### CODE REVIEW

**Review by mschilder123 — APPROVED** [review]

**Comment by korran** [comment]
Do we want to merge this after #389 ?

**Comment by bluegate010** [comment]
> Do we want to merge this after #389 ?

Sure

**Comment by bluegate010** [comment]
> > Do we want to merge this after #389 ?
> 
> Sure

Ready for review.

**Review by rusty1968 — APPROVED** [review]

File: fmc/src/flow/crypto.rs [hunk 1]
```diff
@@ -59,66 +60,74 @@ impl Crypto {
         env.sha384.digest(data)
     }
 
-    /// Calculate HMAC-384
+    /// Calculate HMAC-348 KDF
```
Comment (rusty1968) [inline fmc/src/flow/crypto.rs:63]: NIT : 384.

Comment (bluegate010) [reply fmc/src/flow/crypto.rs]: Thanks, fixed.

File: fmc/src/flow/rt_alias.rs [hunk 2]
```diff
@@ -44,26 +44,25 @@ impl RtAliasLayer {
         hand_off: &mut HandOff,
         input: &DiceInput,
     ) -> CaliptraResult<DiceOutput> {
+        if Self::kv_slot_collides(input.cdi) {
```
Comment (rusty1968) [inline fmc/src/flow/rt_alias.rs:47]: Can collision detection be a compile time assertion?

Comment (bluegate010) [reply fmc/src/flow/rt_alias.rs:47]: Yes, but I think that only works if FMC ignores the key handle number it observes in the handoff table and instead uses a hard-coded value that was compile-time-asserted to be unique from the values used by ROM.

**Review by rusty1968 — APPROVED** [review]

File: fmc/src/flow/rt_alias.rs (hunk 2 above)
Comment (bluegate010) [reply fmc/src/flow/rt_alias.rs:47]: We could make it harder to accidentally introduce collisions though, if we move the key IDs defined [in ROM](https://github.com/chipsalliance/caliptra-sw/blob/2be0e16e6832400053e67069f9f89989a530958c/rom/dev/src/flow/cold_reset/mod.rs#L34-L39) and [in FMC](https://github.com/chipsalliance/caliptra-sw/blob/2be0e16e6832400053e67069f9f89989a530958c/fmc/src/flow/mod.rs#L29-L30) into one file in the common directory. Could take care of this in a follow-up.

Comment (rusty1968) [reply fmc/src/flow/rt_alias.rs:48]: I like the idea.

**Review by rusty1968 — APPROVED** [review]

**Review by mhatrevi — APPROVED** [review]

**Pull request merged**